internal/serverinstall/k8s.go

```diff
@@ -461,16 +465,20 @@ func (i *K8sInstaller) Upgrade(
 		i.Config.Namespace = "default"
 	}
 
+	if i.Config.Timeout == 0 {
+		i.Config.Timeout = 300
+	}
+
 	client := action.NewUpgrade(actionConfig)
 	client.DryRun = false
 	client.DisableHooks = false
 	client.Wait = true
 	client.WaitForJobs = false
 	client.Devel = true
 	client.DependencyUpdate = false
-	client.Timeout = 300 * time.Second
+	client.Timeout = time.Duration(i.Config.Timeout) * time.Second
 	client.Namespace = i.Config.Namespace
-	client.Atomic = false
+	client.Atomic = true
```

Comment:
This should probably be a flag too

Reply:
Should I keep setting it to `true` as the default value? I was discussing about it with @demophoon on #4403

Reply:
I recommend keeping it false by default, because that's the default value in Helm! Might as well match their behavior. At least with the flag it's configurable.